### PR TITLE
fix: 🐛 fix <Select /> for some undefined cases

### PR DESCRIPTION
**Pull request by robertmihaiman**

### CODE REVIEW

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -60,7 +62,7 @@ export const Select: React.FC<SelectProps> = ({
   ...rest
 }) => {
   const initialValue =
-    value || options.find((option) => !option.disabled)?.value || options[0].value;
+    defaultValue || value || options.find((option) => !option.disabled)?.value || options[0].value;
```
Comment (hevele-moda) [inline src/components/Select/Select.tsx:65]: I would give `value` higher priority than `defaultValue`: `value || defaultValue || options.find(...)`.

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -108,10 +110,6 @@ export const Select: React.FC<SelectProps> = ({
     onChange && onChange(state.value);
   }, [onChange, state.value]);
 
-  useUpdateEffect(() => {
-    value && dispatch({ type: 'SELECT', payload: { value } });
-  }, [value]);
```
Comment (hevele-moda) [inline src/components/Select/Select.tsx:113]: Why delete this? I think we still want to be able to control the `value` from props.

Comment (robertmihaiman) [reply src/components/Select/Select.tsx:113]: this hasn't had to get deleted, I don't know what happened

**Pull request merged**